lapixdl/convert.py

```diff
@@ -0,0 +1,251 @@
+        return [coco_pol(geo)]
+    elif geo.geom_type == 'MultiPolygon':
+        return [coco_pol(g) for g in geo.geoms]
+    else:
+        raise TypeError(f'Geometry shape is not a polygon or MultiPolygon. This is a {geo.geom_type}.')
+
+
+def __to_OD_COCO_annotations(
+    lapix_df: Lapix,
+    decimals: int = 2,
+) -> list[dict[str, Any]]:
+    cols = lapix_df.columns
+    if not all(c in cols for c in ['area', 'image_id', 'iscrowd']):
+        raise KeyError('The dataframe need to have the columns `area`, `image_id`, `iscrowd`!')
+
+    return lapix_df.apply(
+        lambda row: {
+            'id': row.name,
+            'image_id': row['image_id'],
+            'category_id': row['category_id'],
+            'bbox': bounds_to_coco_bb(row['geometry'].bounds, decimals),
+            'segmentation': pol_to_coco_segment(row['geometry'], decimals),
+            'area': np.round(row['area'], decimals),
+            'iscrowd': row['iscrowd'],
+        },
+        axis=1,
+    ).to_numpy().tolist()
+
+
+def to_OD_COCO_annotations(
+    lapix_df: Lapix,
+    decimals: int = 2,
+    processes: int = 1
+) -> list[dict[str, Any]]:
+
+    cols = lapix_df.columns
+    if not all(c in cols for c in ['area', 'image_id', 'iscrowd']):
+        raise KeyError('The dataframe need to have the columns `area`, `image_id`, `iscrowd`!')
+
+    # To ensure that will have sequential index
+    lapix_df.reset_index(drop=True, inplace=True)
+    lapix_df.index = lapix_df.index + 1
+
+    # Split equally the annotations for the processes quantity
+    ann_ids_splitted = np.array_split(lapix_df.index.tolist(), processes)
+    print(f'Number of processes: {processes}, annotations per process: {len(ann_ids_splitted[0])}')
+
+    workers = multiprocessing.Pool(processes=processes)
+    procs = []
+    for ann_ids in ann_ids_splitted:
+        df_to_process = lapix_df.loc[lapix_df.index.isin(ann_ids), :]
+        p = workers.apply_async(__to_OD_COCO_annotations, (df_to_process, decimals))
+        procs.append(p)
+
+    annotations_coco = []
+    for p in procs:
+        annotations_coco.extend(p.get())
+
+    return annotations_coco
+
+
+def create_COCO_OD(
+    lapix_df: Lapix,
+    categories_coco: list[dict[str, str]],
+    decimals: int = 2,
+    processes: int = 1,
+    info_coco: dict[str, Any] | None = None
+) -> dict[str, Any]:
+
+    lapix_df_clean = lapix_df.drop_duplicates('image_name')
+
+    images_coco = [
+        {
+            'file_name': row['image_name'],
+            'height': row['image_height'],
+            'width': row['image_width'],
+            'id': row['image_id'],
+        } for _, row in lapix_df_clean.iterrows()
+    ]
+
+    annotations_coco = to_OD_COCO_annotations(lapix_df, decimals, processes)
+
+    coco_od = {
+        'categories': categories_coco,
+        'images': images_coco,
+        'annotations': annotations_coco,
+        'info': info_coco if isinstance(info_coco, dict) else {}
+    }
+
+    return coco_od
```

Comment:
I think it's better to have `x_to_y` functions to improve usability (i.e. `labelbox_to_coco`). Even if it's only a shortcut to `(x) -> lapix_to_y(x_to_lapix(x))`.

Reply:
Done on [3e30be4](https://github.com/lapix-ufsc/lapixdl/pull/34/commits/3e30be492800341fe79636e124089e2425f45d21)

Reply:
Add labelbox_to_coco on [ec63a4d](https://github.com/lapix-ufsc/lapixdl/pull/34/commits/ec63a4d36a1d016cff15ce7a92b123527a448f12)